tasks/bootstrap-firewall.yml

```diff
@@ -11,8 +11,7 @@
     rule: allow
     port: "{{ item }}"
     proto: tcp
-  with_items:
-    - "{{ ports | default(['22']) }}"
+  loop: "{{ sb_debian_base_ports |default(['22']) }}"
```

Comment:
@speedlight since this is a breaking change, I think we need to bump the major version before release

Reply:
@speedlight I think you should be able to declare default values at `defaults/main.ym`:

```yaml
sb_debian_base_ports: ['22']
```

Reply:
@speedlight after a second thought, `ports` or `sb_debian_base_ports` sounds too generic taking into account that we are using these variables mainly to configure the firewall. What do you think about renaming these two variables?

```
sb_debian_base_ports => sb_debian_base_firewall_allowed_tcp_ports
sb_debian_base_ports_ips => sb_debian_base_firewall_allowed_tcp_ips_and_ports
```

In fact, I think we should be able to squash these two variables in one since `src` allows `any` as a wildcard

![image](https://user-images.githubusercontent.com/2049686/78571525-e7e84400-77eb-11ea-948d-fe66ec24b87c.png)

```yaml
sb_debian_base_firewall_allowed_tcp_ports:
  - port: 22
```
Then we could assign a default value at the task level:

```yaml
ufw:
  src: "{{ item.ip | default('any') }}"
  port: "{{ item.port }}"
```

Reply:
@sestrella thanks for the ideas! I think I managed to have a very flexible task.
This is the default value:
```yaml
sb_debian_base_firewall_allowed_ports:
    - port: 22
```
This is the task:
```yaml
- name: Set allowed ports
  ufw:
    rule: allow
    src:  "{{ item.ip |default('any') }}"
    port: "{{ item.port }}"
    proto: "{{ item.proto |default('tcp') }}"
 loop: "{{ sb_debian_base_firewall_allowed_ports }}"
```
This is my test playbook vars:
```yaml
sb_debian_base_firewall_allowed_ports:
  - port: 22
  - port: 161
    proto: udp
    ip: 192.168.168.0/24
  - port: 80
  - port: 443
  - port: 10050:10051
```
Playbook output:
```bash
TASK [sb-debian-base : Set allowed ports] ******************************************************************************************************************************************************************************************************
changed: [localvm] => (item={'port': 22})                                                                                                                                                                                                      
changed: [localvm] => (item={'port': 161, 'proto': 'udp', 'ip': '192.168.168.0/24'})                                                                                                                                                           
changed: [localvm] => (item={'port': 80})                                                                                                                                                                                                      
changed: [localvm] => (item={'port': 443})                                                                                                                                                                                                     
changed: [localvm] => (item={'port': '10050:10051'})
```
And the status on the server:
```bash
Status: active

     To                         Action      From
     --                         ------      ----
[ 1] 22/tcp                     ALLOW IN    Anywhere                  
[ 2] 161/udp                    ALLOW IN    192.168.168.0/24          
[ 3] 80/tcp                     ALLOW IN    Anywhere                  
[ 4] 443/tcp                    ALLOW IN    Anywhere                  
[ 5] 10050:10051/tcp            ALLOW IN    Anywhere                  
[ 6] 22/tcp (v6)                ALLOW IN    Anywhere (v6)             
[ 7] 80/tcp (v6)                ALLOW IN    Anywhere (v6)             
[ 8] 443/tcp (v6)               ALLOW IN    Anywhere (v6)             
[ 9] 10050:10051/tcp (v6)       ALLOW IN    Anywhere (v6)
```
What do you think?

Reply:
> @speedlight since this is a breaking change, I think we need to bump the major version before release, unless, there is a way to create a variable alias and flag the old variable as deprecated. What do you think

Bumping the version could be a better way stating as backward incompatible with previous version and a link to the marker on the readme for how to use the variable. That way the users can easly update their playbooks. WDYT?